DeepMRSeg/deepmrseg_test.py

```diff
@@ -432,140 +461,387 @@ def _main():
 		nJobs = _os.cpu_count()
 	#ENDIF
 
-	# if otherMods defined
+	# Create temp dir
 	#IF
-	if FLAGS.otherMods:
-		otherMods = list(map( str, FLAGS.otherMods.split(',') ))
-		num_modalities = len(otherMods) + 1
+	if FLAGS.tmpDir:
+		user_defined_tmpdir = True
+		tmpDir = FLAGS.tmpDir
+		if not _os.path.isdir(FLAGS.tmpDir):
+			_os.makedirs( FLAGS.tmpDir )
 	else:
-		otherMods = None
-		num_modalities = 1
+		user_defined_tmpdir = False
+		tmpDir = _tempfile.mkdtemp( prefix='deepmrseg_test_' )
 	#ENDIF
 
+	print("Temp folder set to : " + tmpDir)
+	_sys.stdout.flush()
+	
+
 	### Print parsed args
-	print( "\nFile List \t: %s" % (FLAGS.sList) )
-	print( "ID Column \t: %s" % (FLAGS.idCol) )
-	print( "Output Column \t: %s" % (FLAGS.outCol) )
-	print( "Ref Modality \t: %s" % (FLAGS.refMod) )
-	print( "Other Mods \t: %s" % (otherMods) )
-	print( "Num of Mods \t: %d" % (num_modalities) )
+	if IOType == 1:
+		print( "\nInput image(s) \t: %s" % (' , '.join(FLAGS.inImg)) )
+		print( "\nOutput image \t: %s" % (FLAGS.outImg) )
+	if IOType == 2:
+		print( "\nInput dir \t: %s" % (FLAGS.inDir) )
+		print( "\nOutput dir \t: %s" % (FLAGS.outDir) )
+		print( "\nOutput suffix \t: %s" % (FLAGS.outSuff) )
+	if IOType == 3:
+		print( "\nFile List \t: %s" % (FLAGS.sList) )
+
 	print( "Data Prepped \t: %d" % (FLAGS.prep) )
 	
 	print("\nModel Dir(s) \t: %s" % (FLAGS.mdlDir))
-	print( "ROI csv \t: %s" % (FLAGS.roi) )
 
-	print("\nNum of Classes \t: %d" % (FLAGS.num_classes))
-	print("Rescale Method \t: %s" % (FLAGS.rescale))
-	print("XY width \t: %d" % (FLAGS.xy_width))
-	print("Voxel Size \t: %f" % (FLAGS.ressize))
-	print("Orientation \t: %s" % (FLAGS.reorient))
 	print("Batch Size \t: %s" % (FLAGS.batch))
 	
 	print("\nOutput probs \t: %d" % (FLAGS.probs))
 	
-	### Create temp dir, if needed
 	import csv as _csv
 	from concurrent.futures import ThreadPoolExecutor as _TPE
 
+	###############################################################
+	#### For direct I/O options: Create list file with in/out images 
+	if IOType == 1:
+
+		# Generate list of columns
+		numMod = len(FLAGS.inImg)   
+		modCols = 'Mod_'
+		modCols = [modCols + str(i+1) for i in range(numMod)]
+		columns = _np.array(['ID'] + modCols + ['OutImg']).reshape(1,-11)
```

Comment:
What does reshape(1,-11) do here? Should it be reshape(1,-1)?

Reply:
Fixed